pandas/core/series.py

```diff
@@ -4332,6 +4335,65 @@ def between(self, left, right, inclusive=True) -> "Series":
 
         return lmask & rmask
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries, convert_integer: bool = True) -> ABCSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        If the dtype is ``object``, if possible, convert to ``StringDtype``,
+        ``BooleanDtype`` or an appropriate integer extension type, otherwise leave as
+        ``object``.
+
+        If the dtype is "integer", convert to an appropriate integer type.
+
+        If the dtype is numeric, and consists of all integers, convert to an
+        appropriate integer type.
+
+        Parameters
+        ----------
+        convert_integer : bool, default True
+            Whether ``int`` types should be converted to integer extension types
+
+        Returns
+        -------
+        converted : same type as input object
+        """
+        target_int_dtype = "Int64"
+
+        try:
+            inferred_dtype = lib.infer_dtype(self)
+        except ValueError:
+            inferred_dtype = self.dtype
+
+        # If an object, try to convert to an integer, string or boolean
+        # extension type, otherwise leave it alone
+        if is_object_dtype(self.dtype):
+            if (
+                inferred_dtype == "mixed-integer"
+                or inferred_dtype == "mixed-integer-float"
+            ):
+                inferred_dtype = target_int_dtype
+            elif inferred_dtype not in {"string", "boolean", "integer"}:
+                inferred_dtype = self.dtype
+
+        # If an integer, then match the size based on the registry
+        elif is_integer_dtype(self.dtype):
+            if convert_integer and not is_extension_array_dtype(self.dtype):
+                inferred_dtype = _dtypes.get(self.dtype.name, target_int_dtype)
+
+        # If it's not integer and numeric try to make it an integer
+        elif is_numeric_dtype(self.dtype):
+            inferred_dtype = target_int_dtype
+
+        try:
+            result = self.astype(inferred_dtype)
+        except TypeError:
+            result = self.astype(self.dtype)
```

Comment:
Why is this case needed, rather than just passing? It's just to ensure we don't return `self`? In that case, I think `.copy()` more clearly expresses the intent.

The fact that the result is a copy should be asserted in the tests (and perhaps an option should be given for `copy=False`, but that's another matter).

Reply:
done